ethereumj-core/src/main/java/org/ethereum/sync/PeersPool.java

```diff
@@ -193,7 +194,13 @@ public void connect(Node node) {
         }
 
         synchronized (pendingConnections) {
-            ethereum.connect(node);
```

Comment:
Hi Nashatyrev,
I reverted the changes of PeersPool for now and will address this issue in a new pull request.
After looking further into this, my proposal is to create a new class PeerClientManager which would be referenced by any other component which requires peer client connections. I will add a pull request shortly to reflect these changes so you can give me your oppinion on the matter.